src/encoder.rs

```diff
@@ -1208,6 +1211,7 @@ pub fn encode_tx_block<T: Pixel>(
     let tx_dist_scale_bits = 2 * (3 - get_log_tx_scale(tx_size));
     let tx_dist_scale_rounding_offset = 1 << (tx_dist_scale_bits - 1);
     tx_dist = (tx_dist + tx_dist_scale_rounding_offset) >> tx_dist_scale_bits;
+    tx_dist = (fi.dist_scale[0] * tx_dist as f64) as i64
```

Comment:
This function processes a single plane. Perhaps:
```suggestion
    tx_dist = (fi.dist_scale[p] * tx_dist as f64) as i64
```